common/src/main/java/com/microsoft/identity/common/internal/broker/InactiveBrokerClient.java

```diff
@@ -0,0 +1,166 @@
+     */
+    private BrokerAccountServiceFuture connect() throws ClientException {
+
+        if (mBrokerAccountServiceIntent == null) {
+            throw new ClientException("mBrokerAccountServiceIntent is null. BrokerAccountService.");
+        }
+
+        BrokerAccountServiceFuture future = new BrokerAccountServiceFuture();
+        mBrokerAccountServiceConnection = new BrokerAccountServiceConnection(future);
+
+        mBound = mContext.bindService(mBrokerAccountServiceIntent, mBrokerAccountServiceConnection, Context.BIND_AUTO_CREATE);
+        Logger.verbose(TAG + "connect", "The status for MicrosoftAuthService bindService call is: " + Boolean.valueOf(mBound));
+
+        if (!mBound) {
+            throw new ClientException("Service is unavailable or does not support binding. BrokerAccountService.");
+        }
+
+        return future;
+    }
+
+    /**
+     * Disconnects (unbinds) from the bound BrokerAccountService
+     */
+    private void disconnect() {
+        if (mBound) {
+            mContext.unbindService(mBrokerAccountServiceConnection);
+            mBound = false;
+        }
+    }
+
+    /**
+     * Gets the intent that points to the bound service on the device... if available
+     * You shouldn't get this far if it's not available
+     *
+     * @param context
+     * @return Intent
+     */
+    private Intent getIntentForBrokerAccountService(final Context context) {
+        final String inactiveBrokerPackageName = getInactiveBrokerPackageName(context);
+
+        if (inactiveBrokerPackageName == null || inactiveBrokerPackageName.length() == 0) {
+            return null;
+        }
+
+        final Intent authServiceToBind = new Intent(BROKER_ACCOUNT_SERVICE_INTENT_FILTER);
+        authServiceToBind.setPackage(inactiveBrokerPackageName);
+        authServiceToBind.setClassName(inactiveBrokerPackageName, BROKER_ACCOUNT_SERVICE_CLASS_NAME);
+
+        return authServiceToBind;
+    }
+
+    /**
+     * Returns the inactive installed authenticator.
+     * - If CP is the broker, this will return authenticator.
+     * - If authenticator is the broker, this will return CP.
+     *
+     * @param context
+     * @return package name of the installed inactive broker app, if there's any.
+     */
+    @Nullable
+    private String getInactiveBrokerPackageName(@NonNull final Context context) {
+        final String methodName = ":getInactiveBrokerPackageName";
+
+        final String inactiveBrokerPackageName;
+        if (AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equalsIgnoreCase(mCallingPackageName)) {
+            inactiveBrokerPackageName = AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME;
+        } else if (AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(mCallingPackageName)) {
+            inactiveBrokerPackageName = AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME;
+        } else {
+            return null;
```

Comment:
We should never get into this state, but lets log the unknown package name here

Reply:
adding package name logging at the top of the function instead.

Reply:
@rpdome unresolving this as this is still not fixed after the `Resolve comments` commit